src/v/resource_mgmt/cpu_profiler.cc

```diff
@@ -146,4 +149,33 @@ void cpu_profiler::on_sample_period_change() {
     ss::engine().set_cpu_profiler_period(_sample_period());
 }
 
+ss::future<std::vector<cpu_profiler::shard_samples>>
+cpu_profiler::override_and_get_results(
+  std::chrono::milliseconds timeout, std::optional<ss::shard_id> shard_id) {
+    if (_gate.is_closed()) {
+        co_return std::vector<shard_samples>{};
+    }
+    auto holder = _gate.hold();
+
+    auto currently_enabled = is_enabled();
+    _override_enabled++;
+
+    // If there are no other enable overrides currently then signal an enable
+    // change.
+    if (!currently_enabled) {
+        on_enabled_change();
+    }
+
+    co_await ss::sleep(timeout);
```

Comment:
Given how this work (aka we are not streaming yet) does it even make sense to allow `wait_ms` to be bigger than 2 minutes (plus some leeway)?

Reply:
2mins is how long the internal buffers will take to fill with the default sample period. If someone changes that it could take more or less time. 15mins was selected to give some leeway there. I could technically make the upper bound dynamic and have it be the time it'd take to fill the buffers given the current sample rate. Though that may be confusing from an end-users perspective.